lighthouse-core/test/computed/load-simulator-test.js

```diff
@@ -45,14 +53,52 @@ describe('Simulator artifact', () => {
     const throttling = {rttMs: 120, throughputKbps: 1000, cpuSlowdownMultiplier: 3};
     const settings = {throttlingMethod: 'simulate', throttling};
     const context = {settings, computedCache: new Map()};
-    const simulator = await LoadSimulator.request({
-      devtoolsLog,
-      settings,
-    }, context);
+    const simulator = await LoadSimulator.request({devtoolsLog, settings}, context);
 
     assert.equal(simulator._rtt, 120);
     assert.equal(simulator._throughput / 1024, 1000);
     assert.equal(simulator._cpuSlowdownMultiplier, 3);
     assert.equal(simulator._layoutTaskMultiplier, 1.5);
+    simulator.simulate(createNetworkNode());
+
+    const {additionalRttByOrigin, serverResponseTimeByOrigin} = simulator._connectionPool._options;
+    expect(additionalRttByOrigin.get('https://pwa.rocks')).toMatchInlineSnapshot(
+      `0.3960000176447025`
+    );
+    expect(serverResponseTimeByOrigin.get('https://pwa.rocks')).toMatchInlineSnapshot(
+      `159.42199996789026`
+    );
+  });
+
+  it('returns a simulator with precomputed lantern data', async () => {
+    const precomputedLanternData = {
+      additionalRttByOrigin: {
+        'https://pwa.rocks': 1000,
+        'https://www.googletagmanager.com': 500,
+        'https://www.google-analytics.com': 1000,
+      },
+      serverResponseTimeByOrigin: {
+        'https://pwa.rocks': 150,
+        'https://www.googletagmanager.com': 200,
+        'https://www.google-analytics.com': 400,
+      },
+    };
+
+    const settings = {throttlingMethod: 'simulate', precomputedLanternData};
+    const context = {settings, computedCache: new Map()};
+    const simulator = await LoadSimulator.request({devtoolsLog, settings}, context);
+    simulator.simulate(createNetworkNode());
+
+    const {additionalRttByOrigin, serverResponseTimeByOrigin} = simulator._connectionPool._options;
+    expect(additionalRttByOrigin).toEqual(new Map([
+      ['https://pwa.rocks', 1000],
+      ['https://www.googletagmanager.com', 500],
+      ['https://www.google-analytics.com', 1000],
+    ]));
+    expect(serverResponseTimeByOrigin).toEqual(new Map([
+      ['https://pwa.rocks', 150],
+      ['https://www.googletagmanager.com', 200],
+      ['https://www.google-analytics.com', 400],
+    ]));
```

Comment:
since these tests are all very specific to the internal implementation (have to call `simulate()`, options are on the connectionPool, etc), maybe a test somewhere that the precomputed numbers are used when simulated vs what would have resulted from the original graph?

Reply:
good call, done 👍